### PR TITLE
Fix the comparison between hyphens and underscores

### LINKED ISSUE #13519: Content bug: Recommending hypens over underscores in file names
## What page(s) did you find the problem on?

[Getting started with the web > Dealing with files](https://github.com/mdn/content/blob/main/files/en-us/learn/getting_started_with_the_web/dealing_with_files/index.md)

MDN URL: https://developer.mozilla.org/en-US/docs/Learn/Getting_started_with_the_web/Dealing_with_files

## Specific page section or heading?

> ## An aside on casing and spacing
> 
> [...] It's better to separate words with hyphens, rather than underscores: `my-file.html` vs. `my_file.html`.
>
> The short answer is that you should use a hyphen for your file names. The Google search engine treats a hyphen as a word separator but does not regard an underscore that way. For these reasons, it is best to get into the habit of writing your folder and file names lowercase with no spaces and with words separated by hyphens, at least until you know what you're doing. That way you'll bump into fewer problems later down the road.

## What is the problem?  What did you expect to see?

The choice between using hyphens and underscores in file names seems to be a matter of preference, as there are benefits and drawbacks to both: While hyphens are in some cases treated as a word-separator, underscores are easier to read and represent the space character.

I don't think this article should explicitly recommend against using underscores in file names, when many popular websites such as MediaWiki sites (e.g. Wikipedia, Wikia), and especially not when the article itself uses underscores in its own name.

### PR DESCRIPTION
**Pull request by bshramin**
####
#### Related issues

#13519

#### Metadata

- [ ] Adds a new document
- [ ] Rewrites (or significantly expands) a document
- [x] Fixes a typo, bug, or other error

### CODE REVIEW

**Review by wbamberg — COMMENTED** [review]
Thanks for your PR, @bshramin ! I think this needs a little more work.

The new point you've added:

> The choice between using hyphens or underscores to separate words in a file name... is a matter of preference"

... is immediately contradicted by the next paragraph:

> The short answer is that you should use a hyphen for your file names.
 
As discussed in the issue, we ought to explain a bit here about the difference between URL structure and filenames, which this part of the article does not cover. Briefly, I think, we ought to say something like:

* the question of how to name your file is as a matter of personal preference. Common choices are to use lowercase only (or at least to be aware of possible case issues with operating systems) and to use hyphens or underscores instead of spaces. What's most important here is to be consistent.

* the question of how to name _files_ is distinct from the question of how to name URLs. The name you give a file on disk is not necessarily (or even usually) the same name as the name used in its URL. URL structure is generally determined in the configuration of your web server. There _are_ practices you should consider when designing URL structure, in particular regarding how search engines index your pages. For example, these are Google's guidelines: https://developers.google.com/search/docs/advanced/guidelines/url-structure .

@hamishwillee , does that sound like a reasonable outline to you? It's a bit tricky because often they are the same (for instance if you are using file:// URLs or Python's http.server, and those are often going to be the first places where people encounter their own URLs. So i wonder if we ought to try to explain this a bit more...

**Comment by bshramin** [comment]
> Thanks for your PR, @bshramin ! I think this needs a little more work.
> 
> The new point you've added:
> 
> > The choice between using hyphens or underscores to separate words in a file name... is a matter of preference"
> 
> ... is immediately contradicted by the next paragraph:
> 
> > The short answer is that you should use a hyphen for your file names.
> 
> As discussed in the issue, we ought to explain a bit here about the difference between URL structure and filenames, which this part of the article does not cover. Briefly, I think, we ought to say something like:
> 
> * the question of how to name your file is as a matter of personal preference. Common choices are to use lowercase only (or at least to be aware of possible case issues with operating systems) and to use hyphens or underscores instead of spaces. What's most important here is to be consistent.
> * the question of how to name _files_ is distinct from the question of how to name URLs. The name you give a file on disk is not necessarily (or even usually) the same name as the name used in its URL. URL structure is generally determined in the configuration of your web server. There _are_ practices you should consider when designing URL structure, in particular regarding how search engines index your pages. For example, these are Google's guidelines: https://developers.google.com/search/docs/advanced/guidelines/url-structure .
> 
> @hamishwillee , does that sound like a reasonable outline to you? It's a bit tricky because often they are the same (for instance if you are using file:// URLs or Python's http.server, and those are often going to be the first places where people encounter their own URLs. So i wonder if we ought to try to explain this a bit more...

Thank you for the review.
I pushed a new commit trying to fix some of these issues.

File: files/en-us/learn/getting_started_with_the_web/dealing_with_files/index.md [hunk 1]
```diff
@@ -24,12 +24,15 @@ When you are working on a website locally on your computer, you should keep all
...
 
-The short answer is that you should use a hyphen for your file names. The Google search engine treats a hyphen as a word separator but does not regard an underscore that way. For these reasons, it is best to get into the habit of writing your folder and file names lowercase with no spaces and with words separated by hyphens, at least until you know what you're doing. That way you'll bump into fewer problems later down the road.
+The question of how to name files is distinct from the question of how to name URLs. The name you give a file on disk is not necessarily (or even usually) the same as the name used in its URL. URL structure is generally determined in the configuration of your web server. You should consider more strict practices when designing URL structure, particularly how search engines index your pages. For example, these are Google's guidelines: https://developers.google.com/search/docs/advanced/guidelines/url-structure
```
Comment (estelle) [inline files/en-us/learn/getting_started_with_the_web/dealing_with_files/index.md:33]: Can this be simplified down? This is in the "learn", and a basic topic tutorial at that. We should be targeting 3rd-grade reading level. Possibly 8th grade at most.

Comment (wbamberg) [reply files/en-us/learn/getting_started_with_the_web/dealing_with_files/index.md:33]: Yes, I agree. This wasn't supposed to be a proposal for the actual text to include here, just a summary what we ought to be saying.

Comment (bshramin) [reply files/en-us/learn/getting_started_with_the_web/dealing_with_files/index.md:33]: It's no problem. I will try to get it right.

**Comment by bshramin** [comment]
> 

@wbamberg Should I remove the link? or change it to something else?

**Review by wbamberg — COMMENTED** [review]
Sorry to be so slow reviewing this PR, @bshramin .

But the PR doesn't seem to incorporate the review comments at https://github.com/mdn/content/pull/13661#pullrequestreview-903288145.

**Comment by sideshowbarker** [comment]
@bshramin https://github.com/mdn/content/pull/13661#pullrequestreview-961175095 notes the following:

> the PR doesn't seem to incorporate the review comments at https://github.com/mdn/content/pull/13661#pullrequestreview-903288145

Do you plan on updating the PR to address those review comments?

**Comment by sideshowbarker** [comment]
Closing for lack of response

**Pull request closed without merging**